app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/CodeGenerator.kt

```diff
@@ -164,6 +170,19 @@ class CodeGenerator {
         return fileSpec
     }
 
+    /**
+     * Generate the code for the given [Descriptors.Descriptor]. Returns a Boolean determine whether the
+     * descriptor has a property field or not.
+     *
+     * @param fileDescriptor [Descriptors.FileDescriptor] to generate code for.
+     * @return Boolean that represents whether the descriptor has a property field or not.
+     */
+    private fun isHasPropertyField(fileDescriptor: Descriptors.Descriptor): Boolean {
+        return fileDescriptor.fields.any {
+            it.isOptional || it.isRequired || it.isRepeated
```

Comment:
I have a question, in proto3 if `optional int32 x = 1` is considered optional and in proto2 if `required int32 x = 1` is considered required, is this going to return false `int32 x = 1`?

I think we should have a couple of unit tests to verify this doesn't break anything.